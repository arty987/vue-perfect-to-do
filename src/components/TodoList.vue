<script>
import TodoItem from "@/components/TodoItem.vue";

export default {
  name: "TodoList",
  components: {
    TodoItem
  }
}
</script>

<template>
  <TodoItem msg="Meet George"/>
  <TodoItem msg="Hit the gym"/>
  <TodoItem msg="Buy eggs"/>
  <TodoItem msg="Read a book"/>
  <TodoItem msg="Organize office"/>
</template>

<style scoped>

</style>