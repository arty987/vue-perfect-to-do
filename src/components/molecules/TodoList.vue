<script>
import TodoTask from "@/components/molecules/TodoTask.vue";
import {mapGetters} from "vuex";
import draggable from "vuedraggable";

export default {
  name: "TodoList",
  data() {
    return {
      drag: false,
    }
  },
  components: {
    TodoTask,
    draggable
  },
  computed: {
    ...mapGetters([
      "tasks"
    ])
  }
}
</script>

<template>
  <p v-if="!tasks.length" class="text-center text-white">No tasks have been added yet</p>
  <draggable :list="tasks" @start="drag = true" @end="drag = false" item-key="id">
    <template #item="{ element }">
      <TodoTask :task="element"/>
    </template>
  </draggable>
</template>