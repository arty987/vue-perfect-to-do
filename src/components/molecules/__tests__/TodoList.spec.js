import {mount} from '@vue/test-utils'
import TodoList from '@/components/molecules/TodoList.vue';
import {createStore} from "vuex";

describe('TodoList.vue', () => {

    it.each([
            [[]],
            [[
                {
                    id: 1,
                    title: 'Task 1',
                    completed: true,
                },
                {
                    id: 2,
                    title: 'Task 2',
                    completed: false,
                },
                {
                    id: 3,
                    title: 'Task 3',
                    completed: false,
                }
            ]]
        ]
    )('Should render TodoList', (tasks) => {
        const store = createStore({
            getters: {
                tasks: () => tasks
            }
        });

        const wrapper = mount(TodoList, {
            global: {
                plugins: [store]
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });
})