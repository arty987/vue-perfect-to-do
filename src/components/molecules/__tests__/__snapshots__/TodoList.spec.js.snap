// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.vue Should render TodoList 1`] = `
<div
  class="task-container"
>
  <div
    class="no-tasks-list-item"
  >
    No tasks to display
  </div>
  <div />
</div>
`;

exports[`TodoList.vue Should render TodoList 2`] = `
<div
  class="task-container"
>
  <!--v-if-->
  <div>
    <li
      class="list-group-item d-flex justify-content-between align-items-center"
      data-draggable="true"
    >
      <div
        class="d-flex flex-row align-items-center"
      >
        <input
          checked=""
          class="app-checkbox"
          type="checkbox"
        />
        <div
          class="task-title completed"
        >
          Task 1
        </div>
      </div>
      <button
        class="app-button btn-close"
        type="button"
      >
        
        
      </button>
    </li>
    <li
      class="list-group-item d-flex justify-content-between align-items-center"
      data-draggable="true"
    >
      <div
        class="d-flex flex-row align-items-center"
      >
        <input
          class="app-checkbox"
          type="checkbox"
        />
        <div
          class="task-title"
        >
          Task 2
        </div>
      </div>
      <button
        class="app-button btn-close"
        type="button"
      >
        
        
      </button>
    </li>
    <li
      class="list-group-item d-flex justify-content-between align-items-center"
      data-draggable="true"
    >
      <div
        class="d-flex flex-row align-items-center"
      >
        <input
          class="app-checkbox"
          type="checkbox"
        />
        <div
          class="task-title"
        >
          Task 3
        </div>
      </div>
      <button
        class="app-button btn-close"
        type="button"
      >
        
        
      </button>
    </li>
  </div>
</div>
`;
